marklogic-client-api/src/test/java/com/marklogic/client/test/rows/RowManagerValidateDocTest.java

```diff
@@ -0,0 +1,212 @@
+        set.clear();
+        WriteBatcher writeBatcher = dataMovementManager.newWriteBatcher();
+        writeBatcher.addAs("/schematron.sch", new StringHandle("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
+                "<sch:schema xmlns:sch=\"http://purl.oclc.org/dsdl/schematron\">\n" +
+                "  <sch:phase id=\"p1\">\n" +
+                "    <sch:active pattern=\"pt1\"/>\n" +
+                "  </sch:phase>\n" +
+                "  <sch:pattern id=\"pt1\">\n" +
+                "    <sch:rule context=\"Person\">\n" +
+                "      <sch:assert test=\"@Title\">The element Person must have a Title attribute\n" +
+                "\t\t</sch:assert>\n" +
+                "      <sch:assert test=\"count(*) = 2 and count(Name) = 1 and count(Gender) = 1\">The element Person should have the child elements Name and Gender.\n" +
+                "\n" +
+                "\t\t</sch:assert>\n" +
+                "      <sch:assert test=\"*[1] = Name\">The element Name must appear before element Gender.\n" +
+                "\t\t</sch:assert>\n" +
+                "    </sch:rule>\n" +
+                "  </sch:pattern>\n" +
+                "</sch:schema>").withFormat(XML));
+        writeBatcher.flushAndWait();
+
+        // Build the rows to bind to the plan
+        ArrayNode array = mapper.createArrayNode();
+        array.addObject().put("desc", "plug").put("uri", "/validateDoc/doc2.xml");
+        array.addObject().put("desc", "adaptor").put("uri", "/validateDoc/doc1.xml");
+        array.addObject().put("desc", "plug").put("uri", "/validateDoc/doc3.xml");
+        array.addObject().put("desc", "plug").put("uri", "/validateDoc/doc4.xml");
+        array.addObject().put("desc", "adaptor").put("uri", "/validateDoc/doc5.xml");
+        PlanBuilder.Plan plan = op.fromParam("bindingParam", "", op.colTypes(
+                op.colType("uri", "string"),
+                op.colType("desc", "string")
+        ))
+                .validateDoc(op.col("desc"),
+                        op.validateDocSchemaDefinition()
+                                .withKind("schematron")
+                                .withSchemaUri("/schematron.sch"),
+                        op.validateDocErrorDispositionDef().withLogSize(100).withLogLevel("summary"));;
+
+        plan = plan.bindParam("bindingParam", new JacksonHandle(array), null);
+
+        Iterator<RowRecord> rows = rowManager.resultRows(plan).iterator();
+        while (rows.hasNext()){
+            RowRecord rowRecord = rows.next();
+            if(rowRecord.getString("uri") != null)
+                set.add(rowRecord.getString("uri"));
+        }
+        // TODO: uncomment the below after https://bugtrack.marklogic.com/57987 is fixed
+        // assertTrue(set.size() == 5);
+
+    }
+
+    @After
+    public void cleanup(){
+        DocumentManager docMgr = Common.client.newDocumentManager();
+        docMgr.delete("/schema/jsonValidation.json", "/schematron.sch");
```

Comment:
Instead of having to create/delete these, just add them to src/test/ml-schemas - they'll get loaded as part of `mlDeploy`. Just ensure that they'll only apply to docs created in your test.